test/manifest-date.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+const tap = require("tap").test;
+const {
+  generateCatalogueManifest,
+  generateVectorManifest,
+} = require("../scripts/generate-manifest");
+
+const { getExpectedVector } = require("./manifest-v7.14");
+
+const {
+  formatDateToIso,
+  checkDateVersion,
+  coerceToDateSemver,
+} = require("../scripts/date-versions");
+
+const sources = require("./fixtures/valid-sources/sources.json");
+const fieldInfo = require("./fixtures/fieldInfo.json");
+const constants = require("../scripts/constants");
+
+tap("formatDateToIso", (t) => {
```

Comment:
This test is failing for me. I wonder if the `formatDateToIso` function should use the `Date.getUTC*` methods instead? I did a quick test with those methods and the tests ran successfully.

```diff
 FAIL  test/manifest-date.js
 ✖ Render a valid ISO date

  test/manifest-date.js
  25 | tap("formatDateToIso", (t) => {
  26 |   let validDate = "2023-12-31";
> 27 |   t.same(
     | ----^
  28 |     validDate,
  29 |     formatDateToIso(new Date(validDate)),
  30 |     "Render a valid ISO date"

  --- expected
  +++ actual
  @@ -1,1 +1,1 @@
  -"2023-12-30"
  +"2023-12-31"

```

Reply:
Funny that our CI missed this one; thanks for the suggestion. Can you try again with the updated tests from af36903?